daemon/job_actual_ip.go

```diff
@@ -0,0 +1,67 @@
+package daemon
+
+import (
+	"context"
+	"log"
+	"net/netip"
+	"time"
+
+	"github.com/NordSecurity/nordvpn-linux/core"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/network"
+)
+
+func insightsUntilSuccess(ctx context.Context, api core.InsightsAPI) (core.Insights, error) {
+	for i := 0; ; i++ {
+		select {
+		case <-ctx.Done():
+			return core.Insights{}, ctx.Err()
+		default:
+			insights, err := api.Insights()
+			if err == nil && insights != nil {
+				return *insights, nil
+			} else {
+				log.Println(internal.ErrorPrefix, err)
+			}
+
+			backoff := network.ExponentialBackoff(i)
+
+			// Wait before retrying
+			select {
+			case <-time.After(backoff):
+				// Continue to the next retry
+			case <-ctx.Done():
+				return core.Insights{}, ctx.Err() // Exit if context is canceled during sleep
+			}
+		}
+	}
+
+}
+
+func JobActualIP(dm *DataManager, api core.InsightsAPI) func(context.Context, bool) error {
```

Comment:
I see some small inconsistency with the `Job*` name. Other `Job*` functions are actually run by `r.scheduler.NewJob(...)`, but this is not. I was slightly confused why you'll be running infinite loop inside a scheduled job :smile:, but it's not a scheduled job. Maybe would be nicer to have it as a plain function which will be run directly instead of returning function and having `actualIP = JobActualIP(...)` later?